pkg/scheduler/framework/plugins/tainttoleration/taint_toleration.go

```diff
@@ -56,7 +56,7 @@ func (pl *TaintToleration) Name() string {
 // failed by this plugin schedulable.
 func (pl *TaintToleration) EventsToRegister() []framework.ClusterEvent {
 	return []framework.ClusterEvent{
-		{Resource: framework.Node, ActionType: framework.Add | framework.Update},
+		{Resource: framework.Node, ActionType: framework.Add | framework.UpdateNodeTaint},
```

Comment:
suppose the pod is failed by the taint plugin, and no other candidate nodes, now we update the node label for example, and pod could be scheduled on that node, should scheduler have a try for now? I haven't checked the order for those plugins btw.

Reply:
In this scenario, `NodeAffinity` plugin will detect and process the changes:

https://github.com/kubernetes/kubernetes/blob/d14ba948ef63769e9767aebd5a08171832a1bbf6/pkg/scheduler/framework/plugins/nodeaffinity/node_affinity.go#L84-L88

Reply:
Is it a best practice for each plug-in to focus only on its own related events?

Reply:
consider the case that pod is failed by taint plugin, instead of the node affinity, again, I haven't check the order of those plugins.

Reply:
@Huang-Wei implemented this. He can help to review this.

Reply:
This is intentional, please refer to the following issue for more details https://github.com/kubernetes/kubernetes/issues/109437

Reply:
> @Huang-Wei implemented this. He can help to review this.

It's updated in #109442 to be more tolerant of supporting CA workflows.